src/docs/development/platform-integration/add-to-app-android/add-splash-screen.md

```diff
@@ -0,0 +1,221 @@
+short-title: Add a splash screen
+description: Learn how to add a splash screen and launch screen to your Android app.
+---
+
+{% asset
+development/platform-integration/add-to-app-android/add-splash-screen/splash-screens_header.png
+class="mw-100" alt="Add Splash Screen Header" %}
+
+The beginning of a Flutter experience requires a brief wait while Dart
+initializes. Additionally, a full-Flutter app requires a standard Android app
+initialization time, too. Flutter supports the display of a "launch screen"
+while your Android app initializes, and also supports the display of a "splash
+screen" while your Flutter experience initializes. This guide teaches you how to
+use launch screens and splash screens in an Android app with Flutter.
+
+{{site.alert.note}} 
+  Strategies are available to minimize wait time related to Flutter 
+  initialization. Consider [pre-warming a FlutterEngine](), and 
+  [re-using a FlutterEngine throughout your app](), to avoid most wait time.
+{{site.alert.end}} 
+
+## Android Launch Screen
+
+Every Android app requires initialization time while the operating system sets
+up the app's Process. Android provides the concept of a [launch screen] to
+display a `Drawable` while the app is initializing.
+
+[launch screen]: {{site.android-dev}}/topic/performance/vitals/launch-time#themed
+
+Flutter provides support for displaying an Android launch screen before showing
+a `FlutterActivity`. Here are instructions to display an Android launch screen:
+
+### Define a launch theme
+
+In `styles.xml`, define a theme whose `windowBackground` is set to the
+`Drawable` that should be displayed as the launch screen.
+
+```xml
+<style name="LaunchTheme" parent="@android:style/Theme.Black.NoTitleBar">
+    <item name="android:windowBackground">@drawable/launch_background</item>
+</style>
+```
+
+### Define a normal theme
+
+In `styles.xml`, define a normal theme to be applied to `FlutterActivity` after
+the launch screen is gone. The normal theme background only shows for a very
+brief moment after the splash screen disappears, and during orientation change
+and `Activity` restoration. Therefore, it is recommended that the normal theme
+use a solid background color that looks similar to the primary background color
+of the Flutter UI.
+
+```xml
+<style name="NormalTheme" parent="@android:style/Theme.Black.NoTitleBar">
+    <item name="android:windowBackground">@drawable/normal_background</item>
+</style>
+```
+
+### Setup FlutterActivity in AndroidManifest.xml
+
+In `AndroidManifest.xml`, set the `theme` of `FlutterActivity` to the launch
+theme. Then, add a meta-data element to the desired `FlutterActivity` to instruct
+Flutter to switch from the launch theme to normal theme at the appropriate time.
+
+```xml
+<activity
+    android:name=".MyActivity"
+    android:theme="@style/LaunchTheme"
+    // ...
+    >
+    <meta-data
+        android:name="io.flutter.embedding.android.NormalTheme"
+        android:resource="@style/NormalTheme"
```

Comment:
I have the same meta point as above. While reading this, I'm not sure which lines I'm expected to add and which ones have defaults in the create templates and that I just need to modify (and what is the default behavior)

Reply:
It makes sense that once the template has this code that we'd want to reference it here. Is it a good idea for us to predict what will be in those templates and attempt to document it now before that code exists? I'm concerned that it's a great opportunity to have these docs mismatch with the eventual template implementation...if we wait until the template exists to update these guides, then they would be guaranteed to be correct. What do you think?

Reply:
Agree. 

My bad for not realizing this isn't in the tools template yet. 
Let's document what's going to be on the stable we cut for the new android embedding. Until then, let's keep this branch correct for master as the tooling/framework changes.